ignite/metrics/metric.py

```diff
@@ -114,6 +116,19 @@ def _sync_all_reduce(self, tensor: Union[torch.Tensor, numbers.Number]) -> Union
             return tensor.item()
         return tensor
 
+    def ensure_metric_value(self, value: Any, raise_error=True) -> Union[numbers.Number, torch.Tensor]:
```

Comment:
I don’t know if an exception should be raised. Metrics are very generics, it can work with ˋAny`. Metrics associated to loggers should use restricted types. 

@vfdev-5 ?

Reply:
I agree metrics are generics. Maybe remove type checking here and move to `Logger` later when necessary?

Reply:
Ok, let’s try this!

Reply:
I just found one more issue, if `Any` type is allowed, shall we deep flatten a dict in this case? Or shall me only care about `dict` contains all number values?

Reply:
IMO, we should not check the output type and raise an error and just keep previous behaviour.

Reply:
I agree. At the moment, we should flat only the result of metrics. IMHO, what is embedded in dict is related to the metric itself.

We would notice in doc that generic result type ˋAny` is a generic purpose but interoperability with others handlers / features implies some restrictions : scalars (number, tensor) and dict of scalars.

What do you think ?